rust/windows-client/src-tauri/src/debug_commands.rs

```diff
@@ -0,0 +1,175 @@
+                    let _ = self.on_error(&Error::LogFileRollError(e));
+
+                    None
+                })
+        }
+    }
+
+    let (layer, handle) = file_logger::layer(std::path::Path::new("."));
+    setup_global_subscriber(layer);
+
+    let device_id = crate::device_id::get();
+
+    let mut session = Session::connect(
+        common_args.api_url,
+        SecretString::from(common_args.token),
+        device_id,
+        CallbackHandler {
+            handle: Some(handle),
+        },
+    )
+    .unwrap();
+
+    tracing::info!("new_session");
+
+    block_on_ctrl_c();
+
+    session.disconnect(None);
+    Ok(())
+}
+
+/// Test encrypted credential storage
+pub fn credentials() -> Result<()> {
+    use keyring::Entry;
+
+    // TODO: Remove placeholder email
+    let entry = Entry::new_with_target("token", "firezone_windows_client", "username@example.com")?;
+    match entry.get_password() {
+        Ok(password) => {
+            println!("Placeholder password is '{password}'");
+
+            println!("Deleting password");
+            entry.delete_password()?;
+        }
+        Err(keyring::Error::NoEntry) => {
+            println!("No password in credential manager");
+
+            let new_password = "top_secret_password";
+            println!("Setting password to {new_password}");
+            entry.set_password(new_password)?;
+        }
+        Err(e) => return Err(e.into()),
+    }
+
+    Ok(())
+}
+
+/// Test generating a device ID from the BIOS or MAC address. This should survive OS re-installs and uniquely identify a device.
+pub fn device_id() -> Result<()> {
+    use smbioslib::SMBiosSystemInformation as SysInfo;
+
+    let data = smbioslib::table_load_from_device()?;
+    if let Some(uuid) = data.find_map(|sys_info: SysInfo| sys_info.uuid()) {
+        println!("SMBios uuid: {uuid}");
+    } else {
+        println!("SMBios couldn't find uuid");
+    }
+
+    Ok(())
+}
+
+pub use details::wintun;
+
+#[cfg(target_os = "linux")]
+mod details {
+    use crate::prelude::*;
+    pub fn wintun(_: Cli) -> Result<()> {
+        panic!("Wintun not implemented for Linux.");
+    }
+}
+
+#[cfg(target_os = "windows")]
+mod details {
+    use super::*;
+
+    pub fn wintun(_: Cli) -> Result<()> {
+        use std::sync::Arc;
+
+        //Must be run as Administrator because we create network adapters
+        //Load the wintun dll file so that we can call the underlying C functions
+        //Unsafe because we are loading an arbitrary dll file
+        let wintun = unsafe { wintun::load_from_path("./wintun.dll") }?;
+
+        //Try to open an adapter with the name "Demo"
+        let adapter = match wintun::Adapter::open(&wintun, "Demo") {
+            Ok(a) => a,
+            Err(_) => {
+                //If loading failed (most likely it didn't exist), create a new one
+                wintun::Adapter::create(&wintun, "Demo", "Example manor hatch stash", None)?
+            }
+        };
```

Comment:
Apologies for not being able to demo this on my machine, but does this essentially mean that each time the client session starts (i.e. signs in), Admin is needed? Does this prompt the user for anything or is it silent?

Just wondering if we need to look into something like `setcap` here so that we can require admin once on the binary or something and then be done with it in order to improve this UX

Reply:
Will test and add notes on #2751